### PR TITLE
More hub device code cleanup

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/AmqpTransportSettings.cs [hunk 1]
```diff
@@ -19,24 +19,6 @@ public sealed class AmqpTransportSettings : ITransportSettings
         private TimeSpan _operationTimeout;
         private TimeSpan _openTimeout;
 
-        /// <summary>
```
Comment (drwill-ms) [inline iothub/device/src/AmqpTransportSettings.cs:22]: moved properties below ctors

File: iothub/device/src/ConnectionStatusChangeReason.cs [hunk 2]
```diff
@@ -4,7 +4,7 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Connection status change reason supported by DeviceClient
+    /// Connection status change reason supported by the device client.
```
Comment (timtay-microsoft) [inline iothub/device/src/ConnectionStatusChangeReason.cs:7]: Same here
```suggestion
    /// Connection status change reason supported by the client.
```

File: iothub/device/src/MethodInvokeRequest.cs [hunk 3]
```diff
@@ -43,34 +44,38 @@ public MethodInvokeRequest(string methodName, string payload, TimeSpan? response
         }
 
         /// <summary>
-        /// Method to run
+        /// Method to run.
```
Comment (timtay-microsoft) [inline iothub/device/src/MethodInvokeRequest.cs:47]:
```suggestion
        /// Method to invoke.
```

File: iothub/device/src/MethodResponse.cs [hunk 4]
```diff
@@ -55,16 +54,15 @@ private set
...
+        /// Property containing the entire result data, in JSON format.
         /// </summary>
-        public string ResultAsJson => Result == null || Result.Length == 0 ? null : Encoding.UTF8.GetString(Result);
+        public string ResultAsJson => Result == null || Result.Length == 0
+            ? null
+            : Encoding.UTF8.GetString(Result);
 
         /// <summary>
-        /// contains the response of the device client application method handler.
+        /// Contains the response of the device client application method handler.
```
Comment (timtay-microsoft) [inline iothub/device/src/MethodResponse.cs:64]:
```suggestion
        /// The response of the device client application method handler.
```

File: iothub/device/src/MethodResponseStatusCode.cs [hunk 5]
```diff
@@ -4,22 +4,22 @@
 namespace Microsoft.Azure.Devices.Client
 {
     /// <summary>
-    /// Status code for Method Response
+    /// Status code for method response.
     /// </summary>
     public enum MethodResponseStatusCode
```
Comment (timtay-microsoft) [inline iothub/device/src/MethodResponseStatusCode.cs:9]: I'd consider removing this class when we go v2. Users don't need our help defining http level status codes.

**Pull request merged**